Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx

```diff
@@ -306,6 +319,80 @@ export const ProjectTree: React.FC<Props> = ({
     return scope.toLowerCase().includes(filter.toLowerCase());
   };
 
+  const renderTriggerList = (triggers: ITrigger[], dialog: DialogInfo, projectId: string) => {
+    return triggers
+      .filter((tr) => filterMatch(dialog.displayName) || filterMatch(getTriggerName(tr)))
+      .map((tr, index) => {
+        const warningContent = triggerNotSupported(dialog, tr);
+        const errorContent = notificationMap[projectId][dialog.id].some(
+          (diag) => diag.severity === DiagnosticSeverity.Error && diag.path?.match(RegExp(`triggers\\[${index}\\]`))
+        );
+        return renderTrigger(
+          { ...tr, index, displayName: getTriggerName(tr), warningContent, errorContent },
+          dialog,
+          projectId
+        );
+      });
+  };
+
+  const renderTriggerGroupHeader = (groupName: string, dialog: DialogInfo, projectId: string) => {
+    const link: TreeLink = {
+      dialogName: dialog.id,
+      displayName: groupName,
+      isRoot: false,
+      projectId: projectId,
+      skillId: null,
+    };
+    return (
+      <span
+        css={css`
+          margin-top: -6px;
+          width: 100%;
+          label: trigger-group-header;
+        `}
+        role="grid"
```

Comment:
nit: usually used for tables or grid divs.

Reply:
I'm following the TreeItem of the dialog here.  I expect we'll pull this out to a component style in another PR.